tests/unit_tests/regression/test_controller.cpp

```diff
@@ -16,7 +16,9 @@ TEST(ControllerTest, DisconController) {
     // at time = 0.0s
 
     // Use dylib to load the dynamic library and get access to the controller functions
-    const util::dylib lib("./DISCON.dll", util::dylib::no_filename_decorations);
+    const std::filesystem::path project_root = FindProjectRoot();
+    const std::filesystem::path full_path = project_root / "build/tests/unit_tests/DISCON.dll";
```

Comment:
This won't work if your build directory isn't named build.  For example, I always have at least two: build-cpu and build-gpu and make others for different configurations when needed.  It also won't work if the build directory is not located in the project root.

Reply:
I removed usage of `FindProjectRoot()` in the test files by adding a copy of the shared library files in the binary directory for macOS. This is hacky but let's take care of this edge case in a follow-up work and keep making progress to priority items.

Reply:
Sounds good.